app/javascript/mastodon/components/display_name.tsx

```diff
@@ -1,41 +1,40 @@
 import React from 'react';
-import ImmutablePropTypes from 'react-immutable-proptypes';
-import PropTypes from 'prop-types';
-import { autoPlayGif } from 'mastodon/initial_state';
-import Skeleton from 'mastodon/components/skeleton';
-
-export default class DisplayName extends React.PureComponent {
-
-  static propTypes = {
-    account: ImmutablePropTypes.map,
-    others: ImmutablePropTypes.list,
-    localDomain: PropTypes.string,
-  };
+import { autoPlayGif } from '..//initial_state';
+import Skeleton from './skeleton';
+import { Account } from '../../types/resources';
+import { List } from 'immutable';
+
+type Props = {
+  account: Account;
+  others: List<Account>;
+  localDomain: string;
+}
+export default class DisplayName extends React.PureComponent<Props> {
 
-  handleMouseEnter = ({ currentTarget }) => {
+  handleMouseEnter: React.ReactEventHandler<HTMLSpanElement> = ({ currentTarget }) => {
     if (autoPlayGif) {
       return;
     }
 
-    const emojis = currentTarget.querySelectorAll('.custom-emoji');
+    const emojis = currentTarget.querySelectorAll<HTMLImageElement>('.custom-emoji');
```

Comment:
It should be `HTMLElement` here, or you need to change the selector to be `img.custom-emoji` to ensure the element is really an image